app/example_protected_routes.py

```
@@ -74,13 +83,13 @@ async def create_resource(
 
 @router.delete("/delete-resource/{resource_id}")
 async def delete_resource(
-    resource_id: str,
-    current_user: Dict[str, Any] = Depends(require_admin)
+    resource_id: int,  
+    current_user: Dict[str, Any] = Depends(get_current_user)  
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)